scripts/submit_finetune_job.py

```diff
@@ -0,0 +1,202 @@
+    else:
+        wandb_api_key = os.environ.get("WANDB_API_KEY")
+
+    # if config is passed, load and merge that
+    def override_and_reconstruct_command(original_command, train_config, unknown_args):
+        def parse_args(args):
+            cmd_dict = {}
+            i = 0
+            while i < len(args):
+                if args[i].startswith('--'):
+                    key = args[i][2:]
+                    if i + 1 < len(args) and not args[i+1].startswith('--'):
+                        cmd_dict[key] = args[i+1]
+                        i += 2
+                    else:
+                        cmd_dict[key] = True
+                        i += 1
+                else:
+                    i += 1
+            return cmd_dict
+        # Split the original command into a list
+        cmd_parts = shlex.split(original_command)
+
+        # Find the index of open_instruct/finetune.py
+        script_index = cmd_parts.index('open_instruct/finetune.py')
+
+        # Separate the command into pre-script and post-script parts
+        pre_script = cmd_parts[:2]  # 'accelerate launch'
+        pre_script_args = cmd_parts[2:script_index]
+        post_script_args = cmd_parts[script_index+1:]
+
+        # Parse arguments
+        pre_dict = parse_args(pre_script_args)
+        post_dict = parse_args(post_script_args)
+
+        # Combine dictionaries and apply overrides
+        cmd_dict = {**pre_dict, **post_dict}
+        cmd_dict.update(train_config)
+        cmd_dict.update(unknown_args)
+
+        # Reconstruct the command string
+        new_cmd_parts = pre_script + ['open_instruct/finetune.py']
+        for key, value in cmd_dict.items():
+            new_cmd_parts.append(f'--{key}')
+            if value is not True:
+                new_cmd_parts.append(str(value))
+
+        return ' '.join(new_cmd_parts)
+
+    new_arguments = override_and_reconstruct_command(d1['tasks'][0]['arguments'][0], train_config, unknown_args)
+    
+    model_name = get_model_name(new_arguments)
+    # if model name has /, replace with _
+    model_name = model_name.replace("/", "_")
+    # try given config only has one
+    dataset_name, dataset_mixer, train_file = check_dataset_selection(new_arguments)
+    print(f"Dataset selection is valid.")
+    print(f"Dataset name: {dataset_name}")
+    print(f"Dataset mixer: {dataset_mixer}")
+    print(f"Train file: {train_file}")
+
+    d = copy.deepcopy(d1)
+    d['tasks'][0]['arguments'][0] = new_arguments
+
+    # name and description
+    exp_name = f"open_instruct_finetune_{model_name}_{now}"
+    d['description'] = exp_name
+    d['tasks'][0]['name'] = exp_name
+
+    # WANDB settings
+    for env in d['tasks'][0]['envVars']:
+        if env['name'] == "WANDB_DISABLED":
+            env['value'] = False
+        if env['name'] == "WANDB_PROJECT":
+            env['value'] = wandb_project
+    d['tasks'][0]['envVars'].append({
+        'name': 'WANDB_API_KEY', 'value': wandb_api_key
```

Comment:
this might expose the wandb key in the beaker config, rather than set in the workspace as a secret....